File: backend/app/schemas/analytics.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel

class KPICard(BaseModel):
    title: str
    value: str
    change_pct: float
    change_direction: str # "up" or "down"
    subtitle: str
    icon: str

class TimeSeriesPoint(BaseModel):
    date: str
    total_transactions: int
    agentic_transactions: int
    ai_assisted_transactions: int
    human_only_transactions: int
    agentic_adoption_rate: float # percentage e.g. 57.3
    avg_decision_time_sec: float
    budget_savings_usd: float

class CategoryAdoption(BaseModel):
    category: str
    agentic_share_pct: float
    total_volume: int
    growth_yoy_pct: float

class AnalyticsDashboardResponse(BaseModel):
    kpis: List[KPICard]
    time_series: List[TimeSeriesPoint]
    category_breakdown: List[CategoryAdoption]
    efficiency_metrics: Dict[str, Any]

File: frontend/src/types/analytics.ts
export interface KPICard {
  title: string;
  value: string;
  change_pct: number;
  change_direction: 'up' | 'down';
  subtitle: string;
  icon: string;
}

export interface TimeSeriesPoint {
  date: string;
  total_transactions: number;
  agentic_transactions: number;
  ai_assisted_transactions: number;
  human_only_transactions: number;
  agentic_adoption_rate: number;
  avg_decision_time_sec: number;
  budget_savings_usd: number;
}

export interface CategoryAdoption {
  category: string;
  agentic_share_pct: number;
  total_volume: number;
  growth_yoy_pct: number;
}

export interface AnalyticsDashboardResponse {
  kpis: KPICard[];
  time_series: TimeSeriesPoint[];
  category_breakdown: CategoryAdoption[];
  efficiency_metrics: Record<string, any>;
}
